src/circuit.rs

```diff
@@ -2,61 +2,759 @@
+        let (sinsemilla_config_1, merkle_config_1) = {
+            let sinsemilla_config_1 = SinsemillaChip::configure(
+                meta,
+                advices[..5].try_into().unwrap(),
+                advices[6],
+                lagrange_coeffs[0],
+                lookup,
+                range_check.clone(),
+            );
+            let merkle_config_1 = MerkleChip::configure(meta, sinsemilla_config_1.clone());
+
+            (sinsemilla_config_1, merkle_config_1)
+        };
+
+        // Configuration for a Sinsemilla hash instantiation and a
+        // Merkle hash instantiation using this Sinsemilla instance.
+        // Since the Sinsemilla config uses only 5 advice columns,
+        // we can fit two instances side-by-side.
+        let (sinsemilla_config_2, merkle_config_2) = {
+            let sinsemilla_config_2 = SinsemillaChip::configure(
+                meta,
+                advices[5..].try_into().unwrap(),
+                advices[7],
+                lagrange_coeffs[1],
+                lookup,
+                range_check,
+            );
+            let merkle_config_2 = MerkleChip::configure(meta, sinsemilla_config_2.clone());
+
+            (sinsemilla_config_2, merkle_config_2)
+        };
+
+        // Configuration to handle decomposition and canonicity checking
+        // for CommitIvk.
+        let commit_ivk_config =
+            CommitIvkConfig::configure(meta, advices, sinsemilla_config_1.clone());
+
+        // Configuration to handle decomposition and canonicity checking
+        // for NoteCommit_old.
+        let old_note_commit_config =
+            NoteCommitConfig::configure(meta, advices, sinsemilla_config_1.clone());
+
+        // Configuration to handle decomposition and canonicity checking
+        // for NoteCommit_new.
+        let new_note_commit_config =
+            NoteCommitConfig::configure(meta, advices, sinsemilla_config_2.clone());
+
+        Config {
+            primary,
+            q_orchard,
+            q_add,
+            advices,
+            ecc_config,
+            poseidon_config,
+            merkle_config_1,
+            merkle_config_2,
+            sinsemilla_config_1,
+            sinsemilla_config_2,
+            commit_ivk_config,
+            old_note_commit_config,
+            new_note_commit_config,
+        }
     }
 
     fn synthesize(
         &self,
-        _config: Self::Config,
-        _layouter: impl Layouter<pallas::Base>,
+        config: Self::Config,
+        mut layouter: impl Layouter<pallas::Base>,
     ) -> Result<(), plonk::Error> {
+        // Load the Sinsemilla generator lookup table used by the whole circuit.
+        SinsemillaChip::load(config.sinsemilla_config_1.clone(), &mut layouter)?;
+
+        // Construct the ECC chip.
+        let ecc_chip = config.ecc_chip();
+
+        // Witness private inputs that are used across multiple checks.
+        let (rho_old, psi_old, cm_old, g_d_old, ak, nk, v_old, v_new) = {
```

Comment:
```suggestion
        let (psi_old, rho_old, cm_old, g_d_old, ak, nk, v_old, v_new) = {
```
Just putting the tuple in the same order as the values are witnessed.